libs/blockchain/src/blockchain-rpc-query.service.ts

```diff
@@ -252,6 +277,10 @@ export class BlockchainRpcQueryService extends PolkadotApiService {
     return (await api.query.capacity.epochLength()).toNumber();
   }
 
+  public async getCapacityCostForExt(enocdedExt: HexString): Promise<FeeDetails> {
+    return this.api.rpc.frequencyTxPayment.computeCapacityFeeDetails(enocdedExt, null);
+  }
+
```

Comment:
I've been trying to isolate clients of this class from having to decode SCALE; basically redefining all interfaces in terms of regular JS types and exporting those interfaces at the top of the file. So you would export an `IFeeDetails` interface. Each method is responsible for converting its results and returning them--either returning `value | null`, or `value` (with some default for an empty response), or throwing an error, as appropriate.
 
 Up to you, but that's the pattern this class was developed with. Also makes it much easier to mock in unit tests. See the `capacityInfo()` method and corresponding `ICapacityInfo` interface for an example.

Reply:
I like that design choice, will follow through the example :+1:

Reply:
Sounds good--while you're at it, I suppose it's worth putting a comment header at the top of this file explaining that design choice for future maintainers.

Reply:
👍🏻 
Mind putting in a file header comment like I mentioned above? Probably should go in all 3 blockchain class files: `PolkadotApiService`, `BlockchainRpcQueryService`, and `BlockchainService`. Maybe:

```typescript
/*
 * NOTE: This class is designed to isolate consumers from having to deal with the details of interacting directly
 *       with the Frequency blockchain. To that end, return values of functions should not expose the SCALE-
 *       encoded objects that are directly returned from Frequency RPC calls; rather, all payloads should be
 *       unwrapped and re-formed using native Javascript types.
 *
 *       RPC methods that have the potential to return values wrapped as `Option<...>` or any value supporting
 *       the `.isEmpty`, `.isSome`, or `.isNone` getters should implement one of the following behaviors:
 *          - have a specified return type of `<type> | null` and return null for an empty value
 *          - return some sane default for an empty value
 *          - throw an error if an empty value is encountered
 */
 ```